lib/HTFeed/PackageType/Simple/Download.pm

```diff
@@ -0,0 +1,73 @@
+package HTFeed::PackageType::Simple::Download;
+
+use warnings;
+use strict;
+use base qw(HTFeed::Stage::Download);
+use HTFeed::Config qw(get_config);
+use Log::Log4perl qw(get_logger);
+
+my $download_base = "dropbox:";
+
+sub run {
+  my $self = shift;
+
+  my $volume = $self->{volume};
+  # check if the item already exists at $volume->get_sip_location()
+  my $sip_loc = $volume->get_sip_location();
+  if (-f $sip_loc) {
+    get_logger->trace("$sip_loc already exists");
+  } else {
+    my $rclone_config_path = get_config('rclone_config_path');
+    unless (defined $rclone_config_path) {
+      $self->set_error('StageFailed', detail => "rclone_config_path not configured");
+      return 0;
+    }
+    my $nspkg = $volume->get_nspkg();
+    my $dropbox_folder;
+    eval { 
```

Comment:
hmm, there should be a way to check this without throwing an exception if it isn't there. Maybe we just set it to `undef` by default in `HTFeed::Namespace`?

Reply:
Is stage failure nonetheless the expected behavior if there is no dropbox folder configured for the namespace?

Reply:
If the file doesn't exist and the dropbox folder isn't configured, then yes, it should fail with a message to that effect.

Reply:
The only "clean" way I see to modify `HTFeed::Namespace` is to add a guard clause in the `get` inner exception handler (https://github.com/hathitrust/feed/blob/master/lib/HTFeed/Namespace.pm#L76), since the logic there is using `if (defined ...)` on everything: initializing `dropbox_folder => undef` will just fall through to a `croak` there anyway. So I think I'm gonna keep the `eval` bit unless there's an avenue I haven't explored.

Maybe adding a non-exception-raising "has_key" method akin to Ruby `key?` in `HTFeed::Namespace` would restore some order to the insanity?

Reply:
Hmmm.. Maybe what we want is `exists` instead of `defined` so we don't end up throwing an exception when something is present but `undef`; I very much doubt we're relying anywhere on the existing behavior. I'll push a commit with the proposed change.

Reply:
I pushed a commit with some cleanup here; tests should pass.